core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/multitenancy/utils/TenantAxisUtils.java

```diff
@@ -455,8 +455,16 @@ public static void cleanupTenants(long tenantIdleTimeMillis) {
                                 PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
                                 carbonContext.setTenantDomain(tenantDomain, true);
 
-                                // Terminating idle tenant configuration contexts.
-                                terminateTenantConfigContext(tenantCfgCtx);
+                                if (carbonContext.getTenantId() != MultitenantConstants.INVALID_TENANT_ID) {
```

Comment:
Shall we invert this if block to avoid NPEs?

```suggestion
                                if (MultitenantConstants.INVALID_TENANT_ID != carbonContext.getTenantId()) {
```